boardswarm-cli/src/main.rs

```diff
@@ -477,42 +492,33 @@ enum Command {
     },
 }
 
-async fn run_configure(
+async fn run_init(
     mut config: config::Config,
     instance: Option<String>,
-    opts: &ConfigureArg,
+    init_args: AuthArg,
 ) -> anyhow::Result<()> {
-    if opts.new && instance.is_none() {
-        bail!("New instances require an instance argument");
+    let mut is_default = false;
+    let instance = instance.unwrap_or_else(|| {
+        is_default = true;
+        "default".to_string()
+    });
+    if let Some(s) = config.find_server(&instance) {
+        bail!("Authentication for {} already exists.", s.name);
     }
 
-    // UGGGG
+    let uri = init_args
+        .uri
+        .as_ref()
+        .ok_or_else(|| anyhow!("New authentication needs a uri"))?;
```

Comment:
If you split the argument structs for `init` and `modify` then clap can already enforce this for you (make url not an Option)

Reply:
ok